packages/material-ui/src/Step/Step.test.js

```diff
@@ -91,60 +49,88 @@ describe('<Step />', () => {
 
   describe('rendering children', () => {
     it('renders children', () => {
-      const { getByTestId } = render(
-        <Step data-testid="root" label="Step One" index={1} orientation="horizontal">
-          <StepChildDiv data-testid="child">Hello World</StepChildDiv>
+      const { container } = render(
+        <Step>
+          <StepButton />
+          <StepLabel />
         </Step>,
       );
 
-      expect(within(getByTestId('root')).getByTestId('child')).not.to.equal(null);
+      const stepLabel = container.querySelector(`.${stepLabelClasses.root}`);
+      const stepButton = container.querySelector(`.${stepButtonClasses.root}`);
+      expect(stepLabel).not.to.equal(null);
+      expect(stepButton).not.to.equal(null);
     });
 
-    it('renders children with all props passed through', () => {
-      const { getAllByTestId } = render(
-        <Step active={false} completed disabled index={0} orientation="horizontal">
-          <PropsAsDataset />
-          <PropsAsDataset />
+    it('should handle null children', () => {
+      const { container } = render(
+        <Step>
+          <StepButton />
+          {null}
         </Step>,
       );
-      getAllByTestId('props').forEach((child) => {
-        // HTMLElement.dataset is a DOMStringMap which fails deep.equal
-        const datasetAsObject = { ...child.dataset };
-        expect(datasetAsObject).to.deep.equal({
-          // props passed from Step
-          active: 'false',
-          alternativeLabel: 'undefined',
-          completed: 'true',
-          disabled: 'true',
-          expanded: 'false',
-          last: 'undefined',
-          icon: '1',
-          orientation: 'horizontal',
-          // test impl details
-          testid: 'props',
-        });
-      });
+
+      const stepButton = container.querySelector(`.${stepButtonClasses.root}`);
+      expect(stepButton).not.to.equal(null);
+    });
+  });
+
+  describe('overriding context props', () => {
+    it('overrides "active" context value', () => {
+      const { getByText } = render(
+        <Stepper activeStep={1}>
+          <Step>
+            <StepLabel>Step 1</StepLabel>
+          </Step>
+          <Step>
+            <StepLabel>Step 2</StepLabel>
+          </Step>
+          <Step active>
+            <StepLabel>Step 3</StepLabel>
+          </Step>
+        </Stepper>,
+      );
+
+      const stepLabel = getByText('Step 3');
+      expect(stepLabel).to.have.class(stepLabelClasses.active);
     });
 
-    it('honours children overriding props passed through', () => {
-      const { getByTestId } = render(
-        <Step active label="Step One" orientation="horizontal" index={0}>
-          <PropsAsDataset active={false} />
-        </Step>,
+    it('overrides "completed" context value', () => {
```

Comment:
What's the use case here for being on a step while the next ones are already completed?

Reply:
It could be about editing a previous step? https://material-ui.com/components/steppers/#non-linear

Reply:
The test isn't about non-linear steppers though.